### PR TITLE
Added Elasticsearch example

### LINKED ISSUE #239: Create an Elasticsearch example

### PR DESCRIPTION
**Pull request by JasonStoltz**
## Description

This PR adds an example project which demonstrates using  Search UI with Elasticsearch.

As noted in the README and elsewhere, when using Elasticsearch as a Search API, no connector is available, so we need to write Handlers ourselves.

This takes a bit more effort than using a Connector, but it provides a lot of flexibility. This example should give Elasticsearch users a bit of a head start when using Search UI.

This example is also deployed as a demo to Netlify, and uses an Elasticsearch instance proxied through an Netlify Function / AWS Lambda. I've noted in the README that proxying Elasticsearch is a best practice.

https://search-ui-stable-elasticsearch.netlify.com

## List of changes

- Added Elasticsearch example
- Updated READMEs to include links to the demo

## Associated Github Issues

Fixes #239

### CODE REVIEW

File: examples/elasticsearch/src/serviceWorker.js [hunk 1]
```diff
@@ -0,0 +1,135 @@
+// This optional code is used to register a service worker.
```
Comment (JasonStoltz) [inline examples/elasticsearch/src/serviceWorker.js]: Boilerplate, no need to review.

File: examples/elasticsearch/src/index.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
```
Comment (JasonStoltz) [inline examples/elasticsearch/src/index.js]: Boilerplate, no need to review.

File: examples/elasticsearch/src/logo.svg [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 841.9 595.3">
```
Comment (JasonStoltz) [inline examples/elasticsearch/src/logo.svg]: Boilerplate, no need to review.

File: examples/elasticsearch/src/index.css [hunk 4]
```diff
@@ -0,0 +1,13 @@
+body {
```
Comment (JasonStoltz) [inline examples/elasticsearch/src/index.css]: Boilerplate, no need to review.

File: examples/elasticsearch/src/App.css [hunk 5]
```diff
@@ -0,0 +1,33 @@
+.App {
```
Comment (JasonStoltz) [inline examples/elasticsearch/src/App.css]: Boilerplate, no need to review.

File: examples/elasticsearch/public/manifest.json [hunk 6]
```diff
@@ -0,0 +1,15 @@
+{
```
Comment (JasonStoltz) [inline examples/elasticsearch/public/manifest.json]: Boilerplate, no need to review.

File: examples/elasticsearch/public/index.html [hunk 7]
```diff
@@ -0,0 +1,38 @@
+<!DOCTYPE html>
```
Comment (JasonStoltz) [inline examples/elasticsearch/public/index.html]: Boilerplate, no need to review.

File: examples/elasticsearch/src/App.test.js [hunk 8]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
```
Comment (JasonStoltz) [inline examples/elasticsearch/src/App.test.js]: Boilerplate, no need to review.

File: examples/elasticsearch/README.md [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+## Setup
+
+### Setup Elasticsearch
+
+To get this project working, you'll first need to have an Elasticsearch instance set up and data indexed.
+
+The National Parks data set that this project uses can be found in `/data`. That data should be indexed into an Elasticsearch index named "national-parks".
+
+If you do not yet have Elasticsearch installed, you can download and [install an instance locally](https://www.elastic.co/products/elasticsearch), or
+create a deployment on [Elastic Cloud](https://www.elastic.co/cloud/).
```
Comment (goodroot) [inline examples/elasticsearch/README.md]: :nice:

File: examples/elasticsearch/README.md [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+ELASTICSEARCH_HOST=https://{user}:{password}@{elasticsearch_url} npm start
+```
+
+Open [http://localhost:3000](http://localhost:3000) to view it in the browser.
+
+## Always proxy your Elasticsearch instance
+
+It is **never** recommended to directly query your Elasticsearch index from a browser.
+
+This project includes a [Netlify](https://www.netlify.com/) [Function](https://www.netlify.com/docs/functions/) which proxies your Elasticsearch.
```
Comment (goodroot) [inline examples/elasticsearch/README.md]: Recommend combining these into one link:

`[Netlify Function](https://www.netlify.com/docs/functions/)`

File: examples/elasticsearch/src/buildRequest.js [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+          query: searchTerm,
+          fields: ["title", "description"]
+        }
+      }
+    : { match_all: {} };
+}
+
+/*
+
+  Converts Current Application State to an Elasticsearch request.
```
Comment (goodroot) [inline examples/elasticsearch/src/buildRequest.js]: No need to case Current Application State.
```suggestion
  Converts current application state to an Elasticsearch request.
```

File: examples/elasticsearch/src/buildRequest.js [hunk 12]
```diff
@@ -0,0 +1,121 @@
...
+
+  When implementing an onSearch Handler in Search UI, the handler needs to take the
+  current state of the application and convert it to an API request.
+
+  For instance, there is a "current" property in the application state that you receive
+  in this handler. The "current" property represents the current page in pagination.
+
+  There is no concepts of "pages" in Elasticsearch, so the closest thing we have to this is
```
Comment (goodroot) [inline examples/elasticsearch/src/buildRequest.js]:
```suggestion
  There is no concept of "pages" in Elasticsearch, so the closest thing we have to this is
```

File: examples/elasticsearch/src/buildRequest.js [hunk 13]
```diff
@@ -0,0 +1,121 @@
...
+  the "from" option. This file take the "current" value (the current page), and the "resultsPerPage"
```
Comment (goodroot) [inline examples/elasticsearch/src/buildRequest.js]:
```suggestion
  the "from" option. This file takes the "current" value (the current page), and the "resultsPerPage"
```

File: examples/elasticsearch/src/buildRequest.js [hunk 14]
```diff
@@ -0,0 +1,121 @@
...
+  value (the size of a page), to calculate the "from" offset for Elasticsearch.
```
Comment (goodroot) [inline examples/elasticsearch/src/buildRequest.js]: I don't reallyyy understand what this is trying to say. We might need to unpack this more and think of a different way to describe the "from" offset.

File: examples/elasticsearch/src/buildRequestFilter.js [hunk 15]
```diff
@@ -0,0 +1,89 @@
+function getTermFilterValue(field, fieldValue) {
+  // We do this because if the value is a boolean value, we need to apply
+  // our filter differently. We're also only storing the string representation
+  // of the boolean value, so we need to convert it to a Boolean.
+
+  // TODO We need better approach for boolean values
```
Comment (goodroot) [inline examples/elasticsearch/src/buildRequestFilter.js]: Hehe.

File: examples/elasticsearch/src/buildState.js [hunk 16]
```diff
@@ -0,0 +1,74 @@
...
+      ])
+      .reduce(addEachKeyValueToObject, {});
+  });
+}
+
+/*
+  Converts an Elasticsearch response to new Application State
+
+  When implementing an onSearch Handler in Search UI, the handler needs to convert
+  search results into new application state that Search UI understands.
```
Comment (goodroot) [inline examples/elasticsearch/src/buildState.js]:
```suggestion
  search results into a new application state that Search UI understands.
```

File: examples/elasticsearch/src/buildState.js [hunk 17]
```diff
@@ -0,0 +1,74 @@
...
+      .map(([fieldName, fieldValue]) => [
+        fieldName,
+        toObject(fieldValue, getHighlight(record, fieldName))
+      ])
+      .reduce(addEachKeyValueToObject, {});
+  });
+}
+
+/*
+  Converts an Elasticsearch response to new Application State
```
Comment (goodroot) [inline examples/elasticsearch/src/buildState.js]:
```suggestion
  Converts an Elasticsearch response to new application state
```

**Comment by JasonStoltz** [comment]
@goodroot Hey Kellen, just FYI, I made the changes you requested, mind having another look?

**Pull request merged**